pkg/cmd/kind/test/internal/testcmd/testcmd.go

```diff
@@ -0,0 +1,363 @@
+	case "bazel":
+		if err := exec.InheritOutput(
+			exec.Command("bazel", "build", "//test/e2e:e2e.test", "//vendor/github.com/onsi/ginkgo/ginkgo"),
+		).Run(); err != nil {
+			return err
+		}
+	case "docker":
+		if err := exec.InheritOutput(
+			exec.Command("make", "all", `WHAT=cmd/kubectl test/e2e/e2e.test vendor/github.com/onsi/ginkgo/ginkgo`),
+		).Run(); err != nil {
+			return err
+		}
+	default:
+		return errors.Errorf("build mode %q is not implemented yet!", buildType)
+	}
+
+	// In Kubernetes's CI: attempt to release some memory after building
+	if len(os.Getenv("KUBETEST_IN_DOCKER")) > 0 {
+		_ = exec.Command("sync").Run()
+		_ = exec.Command("/proc/sys/vm/drop_caches").SetStdin(strings.NewReader("1")).Run()
+	}
+
+	return nil
+}
+
+func checkAlreadyExists(provider *cluster.Provider, name string) error {
+	n, err := provider.ListNodes(name)
+	if err != nil {
+		return err
+	}
+	if len(n) != 0 {
+		return fmt.Errorf("node(s) already exist for a cluster with the name %q", name)
+	}
+	return nil
+}
+
+func testGinkgo(logger log.Logger, provider *cluster.Provider, flags *flagpole, details Details) error {
+	// construct the cluster config
+	config := &v1alpha4.Cluster{
+		Networking: v1alpha4.Networking{
+			IPFamily: v1alpha4.ClusterIPFamily(flags.IPFamily),
+		},
+		Nodes: []v1alpha4.Node{
+			{
+				Role:  v1alpha4.ControlPlaneRole,
+				Image: flags.ImageName,
+			},
+		},
+	}
+	// add e2e workers
+	numNodesForE2E := 2
+	for i := 0; i < numNodesForE2E; i++ {
+		config.Nodes = append(config.Nodes, v1alpha4.Node{
+			Role:  v1alpha4.WorkerRole,
+			Image: flags.ImageName,
+		})
+	}
+
+	// create the cluster
+	logger.V(0).Infof("Creating cluster %q ...\n", flags.Name)
+	if err := provider.Create(
+		flags.Name,
+		cluster.CreateWithV1Alpha4Config(config),
+		cluster.CreateWithRetain(true),
+		cluster.CreateWithWaitForReady(time.Minute),
+		cluster.CreateWithKubeconfigPath(flags.kubeconfigPath()),
+		cluster.CreateWithDisplayUsage(false),
+		cluster.CreateWithDisplaySalutation(false),
+	); err != nil {
+		if errs := errors.Errors(err); errs != nil {
+			for _, problem := range errs {
+				logger.Errorf("%v", problem)
+			}
+			return errors.New("aborting due to invalid configuration")
+		}
+		return errors.Wrap(err, "failed to create cluster")
+	}
+
+	// In Kubernetes CI we don't have real IPv6 connectivity so we need to
+	// employ some work arounds when testing IPv6 clusters
+	// We do this all environments for consistency.
+	if config.Networking.IPFamily == v1alpha4.IPv6Family {
+		if err := enableIPv6Workarounds(logger, provider, flags.Name); err != nil {
+			return err
+		}
+	}
+
+	// run tests
+	skip := details.Skip
+	if details.Parallel {
+		skip = strings.Join([]string{`\[Serial\]`, skip}, "|")
+	}
+	cmd := exec.Command(
+		"./hack/ginkgo-e2e.sh",
+		// don't use the horrid e2e.test """provider""" concept
+		"--provider=skeleton",
+		// need to inform tests of worker count
+		"--num-nodes=", strconv.Itoa(numNodesForE2E),
```

Comment:
the jobs are failing with
> I0118 01:25:13.095] [1] invalid value "" for flag -num-nodes: strconv.ParseInt: parsing "": invalid syntax
I0118 01:25:13.095] [1] Usage of /go/src/k8s.io/kubernetes/bazel-bin/test/e2e/e2e.test:
🤷‍♂

Reply:
ok, there is the problem, is adding an space
```
W0118 01:25:14.472] Call stack:
W0118 01:25:14.472]   1: ./hack/ginkgo-e2e.sh:143 main(...)
W0118 01:25:14.472] Exiting with status 1
W0118 01:25:22.159] ERROR: command "./hack/ginkgo-e2e.sh --provider=skeleton --num-nodes= 2 --report-dir /workspace/_artifacts --disable-log-dump=true --ginkgo.focus '' --ginkgo.skip '\[Serial\]|'" failed with error: exit status 1
```

Reply:
no, the space is a rendering of passing this to the shell.
the problem is that we're passing it as two arguments, but the first contains `=`

marked the PR WIP, I haven't even run this yet. by inspection it's mostly done but it's still a prototype. I'm going to rework things a bit before un-WIP-ing it.